### ISSUE TITLE
Make tests pass on the updated testing infrastructure

### ISSUE THREAD

**Issue by borekb** [versionpress/versionpress#1400]
#1389 improves the testing infrastructure and it seems that with enough "luck", by which I mean a good order of tests or their combination, all tests even pass. However, in practice, it's not very repeatable or reliable.

This issue is about getting the tests run reasonably reliably on both Mac and Windows. There are still debt items mentioned in #1284 like the "temporarily" disabled [`ActivationDeactivationTest`](https://github.com/versionpress/versionpress/blob/e17f0602ffe49679517a2921e4edf34cfe9b1c46/plugins/versionpress/tests/Selenium/ActivationDeactivationTest.php#L14) and [`ThemeCustomizerTest`](https://github.com/versionpress/versionpress/blob/f4a00328d7903ac1c7bcc10673ce5dc621e23e9f/plugins/versionpress/tests/Selenium/ThemeCustomizerTest.php) which are out of scope of this issue.

- [x] Review Docker stack startup & waiting – @pavelevap has reported that some of the tests are reporting errors (`E`, not `F`) as for example database is not ready yet.
    - Resolved by `run-tests.ts`, #1401.
- [x] Make `npm run tests` pass.
    - With improved test runner #1401 and a workaround for `cloneLooksExactlySameAsOriginal` #1410, all tests now pass. 🎉

**Comment by borekb** [comment]
My current experience (on commit b9e6cc356d):

```
$ git checkout b9e6cc356d
$ npm run refresh-dev
$ npm run tests:cleanup
$ npm run tests:full

> versionpress-dev-workspace@1.0.0 tests:full /Users/borekb/dev/VersionPress/versionpress
> docker-compose -f docker-compose-tests.yml run --rm tests-with-wordpress ../vendor/bin/phpunit -c phpunit.xml; npm run tests:stop

Creating network "versionpress_default" with the default driver
Creating volume "versionpress_db-data-for-tests" with default driver
Creating volume "versionpress_wpcli-cache" with default driver
Creating volume "versionpress_wordpress-files" with default driver
Creating volume "versionpress_test-logs" with default driver
Creating versionpress_selenium-hub_1    ... done
Creating versionpress_mysql-for-tests_1 ... done
Creating versionpress_wordpress-for-tests_1 ... done
PHPUnit 5.7.20 by Sebastian Bergmann and contributors.

Runtime:       PHP 7.2.16 with Xdebug 2.6.0
Configuration: /opt/versionpress/tests/phpunit.xml

...............................................................  63 / 738 (  8%)
............................................................... 126 / 738 ( 17%)
............................................................... 189 / 738 ( 25%)
............................................................... 252 / 738 ( 34%)
............................................................... 315 / 738 ( 42%)
............................................................... 378 / 738 ( 51%)
............................................................... 441 / 738 ( 59%)
............................................................... 504 / 738 ( 68%)
............................................................... 567 / 738 ( 76%)
............................................................... 630 / 738 ( 85%)
............S....SS....S..S....S................S....S.S.SSS... 693 / 738 ( 93%)
........S....................S.....FSS......Php Execution Time: 0.033014059066772 Sec
.                   738 / 738 (100%)Bash Execution Time: 0.035650968551636 Sec


Time: 29.63 minutes, Memory: 18.00MB

There was 1 failure:

1) VersionPress\Tests\Workflow\CloneMergeTest::cloneLooksExactlySameAsOriginal
Failed asserting that two strings are equal.
--- Expected
+++ Actual
@@ @@

-		Posted on November 11, 2011Test post for revert
+		Posted on November 11, 2011March 11, 2019Test post for revert
@@ @@

-		Posted on November 11, 2011Test post for revert
+		Posted on November 11, 2011Test post
@@ @@

-		Posted on November 11, 2011March 11, 2019Test post for revert
+		Posted on November 11, 2011Test post for revert
@@ @@

-		Posted on November 11, 2011Test post for revert
+		Posted on November 11, 2011Test post for comments
@@ @@

-		Posted on November 11, 2011Test post
+		Posted on November 11, 2011March 11, 2019Test post

[... Lots of whitespace here, see the attached file for full output ...]

/opt/versionpress/tests/Workflow/CloneMergeTest.php:210
/opt/versionpress/tests/Workflow/CloneMergeTest.php:48

--

There were 16 skipped tests:

[... Omitted ...]

FAILURES!
Tests: 738, Assertions: 950, Failures: 1, Skipped: 16.

> versionpress-dev-workspace@1.0.0 tests:stop /Users/borekb/dev/VersionPress/versionpress
> docker-compose -f docker-compose-tests.yml stop

Stopping versionpress_wordpress-for-tests_1 ... done
Stopping versionpress_selenium-hub_1        ... done
Stopping versionpress_mysql-for-tests_1     ... done
```

Full [`test-output.txt`](https://github.com/versionpress/versionpress/files/2956211/test-output.txt).

When I move Workflow tests to run before End2End tests, everything passes:

<img width="1176" alt="Screenshot 2019-03-12 at 11 00 17" src="https://user-images.githubusercontent.com/101152/54191362-0e875e80-44b6-11e9-9b3d-3147126d2139.png">

![image](https://user-images.githubusercontent.com/101152/54191399-1d6e1100-44b6-11e9-97e6-47b976cef809.png)

It's not "right" but I'm going to commit the change to move us forward.

> **Note**: The tests are trying to find a balance between isolation and performance. We could restore the database and WordPress site before each test but it would make everything extremely slow. Maybe we could restore the DB & site at least before each test suite.

---

Also strange is that the tests now take 30 minutes for me, which is 3x slower than what I experienced a few days ago – https://github.com/versionpress/versionpress/pull/1396#issuecomment-471232739, https://github.com/versionpress/versionpress/issues/1389#issuecomment-471366909. I don't think anything significant changed between then and now but I'll investigate.

**Comment by borekb** [comment]
I no longer understand what's causing the slowdown. I'm trying again on commits 341fec0f0daf05a626763e93e74b45ac5ca79a36 and 29fa9789d7482e1b18ae142c05c8854891de47e5 where I've definitely seen 10-minute runs before and they now both take 30 minutes. 🤔

**Comment by borekb** [comment]
Good is that I'm seeing consistent results on Docker Toolbox on Windows: https://github.com/versionpress/versionpress/pull/1403#issuecomment-473609478.

**Comment by borekb** [comment]
> I no longer understand what's causing the slowdown. I'm trying again on commits [341fec0](https://github.com/versionpress/versionpress/commit/341fec0f0daf05a626763e93e74b45ac5ca79a36) and [29fa978](https://github.com/versionpress/versionpress/commit/29fa9789d7482e1b18ae142c05c8854891de47e5) where I've definitely seen 10-minute runs before and they now both take 30 minutes. 🤔

So I tried `git clean -fxd` && `npm i`, it didn't help.

Then I **restarted Docker** and voila, tests were back at 10 minutes again.

Who will give me those few hours of my life back? 😅

**Comment by borekb** [comment]
With an improved test runner in #1401 and a workaround for a `cloneLooksExactlySameAsOriginal` implemented in #1410, all tests now pass 🎉.
